From 8df8d10edc8909e19e60f0cc1dd65c1fe706ab67 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Tue, 15 Sep 2015 15:50:18 -0500
Subject: [PATCH] MLK-11555: mtd: nand: i.MX6UL supports bitflip detection

i.MX6UL also has the DEBUG1 register which can be used for bitflip
detection for erased page.

Signed-off-by: Han Xu <b45815@freescale.com>
---
 drivers/mtd/nand/gpmi-nand/gpmi-lib.c  | 4 ++--
 drivers/mtd/nand/gpmi-nand/gpmi-nand.c | 3 ++-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/mtd/nand/gpmi-nand/gpmi-lib.c b/drivers/mtd/nand/gpmi-nand/gpmi-lib.c
index b8ce976..9fa6ec4 100644
--- a/drivers/mtd/nand/gpmi-nand/gpmi-lib.c
+++ b/drivers/mtd/nand/gpmi-nand/gpmi-lib.c
@@ -304,8 +304,8 @@ int bch_set_geometry(struct gpmi_nand_data *this)
 			| BF_BCH_FLASH0LAYOUT1_DATAN_SIZE(block_size, this),
 			r->bch_regs + HW_BCH_FLASH0LAYOUT1);
 
-	/* Set erase threshold to gf/2 for mx6qp and mx7 */
-	if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7(this))
+	/* Set erase threshold to gf/2 for mx6ul, mx6qp and mx7 */
+	if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7(this) || GPMI_IS_MX6UL(this))
 		writel(BF_BCH_MODE_ERASE_THRESHOLD(gf_len/2),
 			r->bch_regs + HW_BCH_MODE);
 
diff --git a/drivers/mtd/nand/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/gpmi-nand/gpmi-nand.c
index c09a6f2..21f3c55 100644
--- a/drivers/mtd/nand/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/gpmi-nand/gpmi-nand.c
@@ -1086,7 +1086,8 @@ static int gpmi_ecc_read_page(struct mtd_info *mtd, struct nand_chip *chip,
 			continue;
 
 		if (*status == STATUS_ERASED) {
-			if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7(this))
+			if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7(this) ||
+						GPMI_IS_MX6UL(this))
 				if (readl(bch_regs + HW_BCH_DEBUG1))
 					flag = 1;
 			continue;
-- 
1.8.0

